FifthLesson/VectorProgram/VectorProgram/Model/Vector.cs

```diff
@@ -1,115 +1,96 @@
+            var secondHashCode = SecondCoordinate.GetHashCode();
+            var thirdHashCode = ThirdCoordinate.GetHashCode();
+            
+            var result = firstHashCode;
+            result = unchecked(result * 397) ^ secondHashCode;
+            result = unchecked(result * 397) ^ thirdHashCode;
+            
+            return result;
         }
 
-        public Vector(Vector vector) : this()
+        public override bool Equals(object obj)
         {
-            CoordinateFirst = vector.CoordinateFirst;
-            CoordinateSecond = vector.CoordinateSecond;
-            CoordinateThird = vector.CoordinateThird;
+            if (ReferenceEquals(null, obj)) return false;
+            if (ReferenceEquals(this, obj)) return true;
+            return (this.GetType() != obj.GetType()) ? false : this.Equals((Vector)obj);
         }
-
-        public override string ToString()
+  
+        public bool Equals(Vector other)
         {
-            return $"({CoordinateFirst}, {CoordinateSecond}, {CoordinateThird})";
+            if (ReferenceEquals(null, other)) return false;
+            if (ReferenceEquals(this, other)) return true;
+            var ValueСomparison = FirstCoordinate == other.FirstCoordinate && SecondCoordinate == other.SecondCoordinate && ThirdCoordinate == other.ThirdCoordinate;
+            return ValueСomparison;
         }
-
-        public static Vector operator +(Vector vector1, Vector vector2)
+        
+        public static Vector operator +(Vector first, Vector second)
         {
-            var result = new Vector(vector1);
-
-            result.CoordinateFirst += vector2.CoordinateFirst;
-            result.CoordinateSecond += vector2.CoordinateSecond;
-            result.CoordinateThird += vector2.CoordinateThird;
-
-            return result;
+            return new Vector
+            {
+                FirstCoordinate = first.FirstCoordinate + second.FirstCoordinate,
+                SecondCoordinate = first.SecondCoordinate + second.SecondCoordinate,
+                ThirdCoordinate = first.ThirdCoordinate + second.ThirdCoordinate
+            };
         }
 
-        public static Vector operator -(Vector vector1, Vector vector2)
+        public static Vector operator -(Vector first, Vector second)
         {
-            var result = new Vector(vector1);
-
-            result.CoordinateFirst -= vector2.CoordinateFirst;
-            result.CoordinateSecond -= vector2.CoordinateSecond;
-            result.CoordinateThird -= vector2.CoordinateThird;
-
-            return result;
+            return new Vector
+            {
+                FirstCoordinate = first.FirstCoordinate - second.FirstCoordinate,
+                SecondCoordinate = first.SecondCoordinate - second.SecondCoordinate,
+                ThirdCoordinate = first.ThirdCoordinate - second.ThirdCoordinate
+            };
         }
 
         public static Vector operator *(double numeric, Vector vector)
```

Comment:
I think, it would be better to re-name parameter **numeric** to **number**

Reply:
Corrected